lib/servctrl/cmd.go

```diff
@@ -156,30 +156,33 @@ func (term *ServTerm) startInteraction() {
 			// split line into a header (lineSplit[0]) and message contents (lineSplit[1]) for more robust parsing
 			var lineSplit = strings.SplitN(line, ": ", 2)
 
-			// case where the server is starting
-			if ServStats.Status == "starting" {
-				// for modded server terminal compatibility, use separate check for "[Server thread/INFO]" and flag-word
-
-				// if the terminal contains flag-word "Preparing spawn area:", update ServStats.LoadProgress
-				if strings.Contains(lineSplit[0], "[Server thread/INFO]") && strings.HasPrefix(lineSplit[1], "Preparing spawn area:") {
-					ServStats.LoadProgress = strings.Split(strings.Split(lineSplit[1], "Preparing spawn area: ")[1], "\n")[0]
-				}
-				// if the terminal contains flag-word "Done", the minecraft server is online
-				if strings.Contains(lineSplit[0], "[Server thread/INFO]") && strings.HasPrefix(lineSplit[1], "Done") {
-					ServStats.Status = "online"
-					log.Print("*** MINECRAFT SERVER IS ONLINE!")
-
-					// launch a stopInstance so that if no players connect the server will shutdown
-					RequestStopMinecraftServer()
+			// only parse lines which can be split in two
+			if len(lineSplit) == 2 {
+				// case where the server is starting
+				if ServStats.Status == "starting" {
+					// for modded server terminal compatibility, use separate check for "[Server thread/INFO]" and flag-word
+
+					// if the terminal contains flag-word "Preparing spawn area:", update ServStats.LoadProgress
+					if strings.Contains(lineSplit[0], "[Server thread/INFO]") && strings.HasPrefix(lineSplit[1], "Preparing spawn area:") {
```

Comment:
`strings.HasPrefix(lineSplit[1], "Preparing spawn area:")` should be `strings.HasPrefix(lineSplit[1], "Preparing spawn area: ")` to avoid bugs

Reply:
yep, I see what you mean, fixed.